### ISSUE TITLE
[a11y support] Benefits Decision Review Team: PDF/HTML a11y Feedback

### ISSUE THREAD

**Issue by sara-amanda** [department-of-veterans-affairs/va.gov-team#72174]
## a11y Assignee(s)

- @sara-amanda 
- Evan (no GH yet)
- @coforma-jamie 

## About the Benefits Decision Review Team

**Note**: DK is on this team, so be sure to include him, as he is an a11y. 

<details><summary>Toggle to view team info.</summary>

- **Team name**: Benefits Decision Review
- **OCTO product owner:** Zach Goldfine
- **Product name:** Board Appeal / Notice of Disagreement
- **Product manager:**  Sade Ragsdale
- **Slack channel:** [Benefits-Decision-Review](https://dsva.slack.com/archives/C5AGLBNRK)
- **Dedicated content writer on your team (if you have one):** N/A - Designer - Eileen Chi
- **Dedicated a11y specialist on your team (if you have one):** DK, David Kennedy
</details>

## Background Information/Previous Conversations

Refer to [intake ticket a11y task list](https://github.com/department-of-veterans-affairs/va.gov-team/issues/71952) in addition to notes below, which can be toggled to expose.

<details><summary>Toggle to view background info.</summary>

- [This is connected to the a11y chat the other day with DK FYI](https://dsva.slack.com/archives/C0335Q88X0S/p1702070187818259) in DSVA Slack.
- **Testing** will be in Staging. 
- Confirmation Page and Persistent HTML
### How it works
- **This item was originally meant for central mail,** not for Veterans to receive. 
    - Would be a larger lift to build the PDFs, so trying the HTML version approach first. 
- **Dependency** of lighthouse to make the PDF - so far they have not been accessible, layering data on top of that. 
- Document is unique to each person, based on the data in the system from the Decision Review. 
- Not a static thing we can remediate. 
### PDF Process
- User fills out a form
- That data populates a web page and that populates a generated PDF from Lighthouse. 
    - What it is creating isn’t accessible.
### About the Data
- The data itself is fine.
- 1 data source going to two places. 
- **If the PDF is marked correctly in HTML** - it could be made accessible. 
- **If flattened** there would be no readable text. 
</details>

### Issue Summary
- [Refer to Background Information/Previous Conversations section of this ticket for full details. ](https://github.com/department-of-veterans-affairs/va.gov-team/issues/72174#:~:text=Toggle%20to%20view%20background%20info.)
    - Lighthouse is generating unique PDFs for each person, based on the Decision review data.
    - We can't remediate those PDFs, but they are going to be made available to users and are not accessible.
    - The VFS team is going to make an HTML page as an approach, from this same data source, so the user can have something accessible. 

### a11y Brainstorming Results
Based on the[ background information, previous conversations](https://github.com/department-of-veterans-affairs/va.gov-team/issues/72174#:~:text=Toggle%20to%20view%20background%20info.) and [issue summary,](https://github.com/department-of-veterans-affairs/va.gov-team/issues/72174#:~:text=a11y%20Solution%20Discussion-,Issue%20Summary,-Refer%20to%20Background) CAIA is recommending the following approach to make the PDF accessible, as discussed between Sara, Evan, SK and Jamie.

- **_Utilize HTML page to generate the PDF_** 
    - Can you generate a PDF from the HTML page being tagged **_instead of_** dual creation HTML page and the PDF from the data at the same time? 
- **Upon Request:** 
    - Additional idea from Evan who said we could try and offer a concierge service to make it accessible and remediate, upon request? 
- **Text About Accessibility**
    - You could consider adding text that mentions something along the lines of viewing the HTML page is the best option for users of assistive technology.
    - This way you are not calling it out that it isn't accessible, but directing traffic to the best option for assistive technology. 
### Rationale
- **User Expectations:** Expectation is that it would be accessible.
    - The VA strives to be accessible and has begun making PDFs accessible. 
    - Our unique audience of Veterans (who use AT more than other traditional sites) _may have a level of expectancy with our PDFs_ vs any out there in the wild world wide web? 
    - **They would likely expect a consistent experience.** 

</details>

**Comment by sara-amanda** [comment]
##CAIA Update

Good Afternoon, @davidakennedy @eileen-coforma! Please find our initial notes from our CAIA A11y Sync,[ regarding potential solutions for your PDF/HTML inquiry in the body of this support ticket.](https://github.com/department-of-veterans-affairs/va.gov-team/issues/72174#:~:text=a11y%20Solution%20Discussion) Let us know your thoughts? 

cc: @coforma-jamie @coforma-terry

**Comment by eileen-coforma** [comment]
@sara-amanda @coforma-jamie 

Thank you for providing feedback!

> Utilize HTML page to generate the PDF
> Can you generate a PDF from the HTML page being tagged instead of dual creation HTML page and the PDF from the data at the same time?

Our team considered creating our own accessible PDF in the frontend, but found it to be a much larger lift + duplicative efforts. 

Since the PDF is already being created by Lighthouse (our dependency) for Central Mail, we would rather that they updated their process (which has been raised) to create an accessible PDF that can be used for both Central Mail and Veterans. That way we can ensure the PDF that is submitted is the same one that's returned to the Veteran.

The endpoint for the current PDF already exists, so creating an HTML page will be significantly less effort than setting up PDF creation within our application. We want to move forward with a quick option to test viability and desire before investing more time on the setup (especially since Lighthouse will also be working towards PDF accessibility).

@davidakennedy can explain the explorations he had done in creating accessible PDFs, and communicating it with Lighthouse

> Upon Request:
> Additional idea from Evan who said we could try and offer a concierge service to make it accessible and remediate, upon request?

Can you expand more on this? I'm not sure what a concierge service entails.

> Text About Accessibility
You could consider adding text that mentions something along the lines of viewing the HTML page is the best option for users of assistive technology.

Do you have language suggestions? I'm not sure what the rules are regarding explicitly calling out users of assistive technology.

**Comment by davidakennedy** [comment]
> @davidakennedy can explain the explorations he had done in creating accessible PDFs, and communicating it with Lighthouse

I talked with the Lighthouse team about some of the bigger tenants of an accessible PDF, including PDF tags, adding a title, setting a language and more. The team did extensive research into the possibilities given its current tech stack, which is Ruby. None of the well known PDF libraries in Ruby create accessible PDFs. Both the team and I looked at at least four major libraries. However, the Lighthouse team does have a potential solution, using another piece of software as a kind of middleware to help in the creation of an accessible PDF using Ruby/Java. The team hopes to work it into its future plans.

> Text About Accessibility
> You could consider adding text that mentions something along the lines of viewing the HTML page is the best option for users of assistive technology.

We talked about this some as a team and with our stakeholders. We definitely had concerns about putting any language about the PDF being "not accessible" for two reasons:

1. It would open up the VA to legal issues.
2. Some Veterans may not know what "not accessible" type language might mean. Our audience is more than Veterans who use screen readers/assistive technology. Some might think "not accessible" means something else.

I like the idea of the suggestion of the "best option" language. We're hoping that research will help reveal what Veterans expect on the page and what they want to do with the information. This way, we can change the headings on the page or other content to make it clear that they can access the information without the PDF. I hope we can do that without calling out shortcomings with the PDF. Also, one thing that @AngelaFowler82 brought up was that many folks who use assistive technology may assume the PDF isn't accessible anyway.

**Comment by coforma-jamie** [comment]
We met with the team today to discuss. 

Future potential workaround in development: Goal is to make the PDFs accessible (via Lighthouse - however, we are not aware of Lighthouse's timeline) - They will replace the feature when it is available. This will impact three of the team's forms positively for accessibility. 

CAIA will work with the team to review the product in staging and assist with user research sessions.

**Comment by sara-amanda** [comment]
^ @coforma-jamie I made a few updates to the [comment above](https://github.com/department-of-veterans-affairs/va.gov-team/issues/72174#issuecomment-1866756886).
